JacksonDunstanNativeCollections/NativeElement.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+
+/// <summary>
+/// NativeElement is a wrapper for a single blittable element to be passed between managed and unmanaged code in jobs
+/// </summary>
+/// <typeparam name="T">The blittable type to be stored</typeparam>
+[StructLayout(LayoutKind.Sequential)]
+[NativeContainer]
+[DebuggerDisplay("{Element}")]
+public unsafe struct NativeElement<T> : IDisposable where T : unmanaged
+{
+    // The pointer to the element
+    [NativeDisableUnsafePtrRestriction]
+    T* m_ptr;
```

Comment:
Fields should be named like `m_MyVar` for consistency with Unity-mandated field names like `m_MinIndex`.

Reply:
Rename to `m_Buffer` for consistency with `NativeIntPtr` and to support `[NativeContainerSupportsDeallocateOnJobCompletion]`.